text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+* In the case of an individual developer signing from his desktop, Sigstore will embed the email address of the signing individual in the metadata delivered to the Rekor service. 
+* In the case of an NPE signer the metadata delivered to the Rekor service will have an Issuer attribute that looks something like this: "Issuer": "https://token.actions.githubusercontent.com".
+
+## Authorization
+
+With respect to AuthZ, while there has been discussion in the pre-rfc thread regarding TUF, this proposal asserts that two phases will need to be undertaken. Phase one will rely on the existing mechanisms implemented on top of git and provided by the most commonly used platforms (GitHub, GitLab, etc, etc) that assign OSS project participants to a group, i.e. “Maintainers” or “Members” in the case of GH. Only project maintainers are typically allowed to either trigger a build event manually or implement automation to do so based on a set of project specific evaluation criteria. We choose this approach to AuthZ as it will allow focus on the core Sigstore implementation for cargo and crates.io in the near term, and a deliberate focus on TUF in the next phase. This is important as TUF is tricky to implement well and represents a significant shift from the existing AuthZ approach that is heavily socialized/embedded broadly in the OSS ecosystem currently.
+
+This RFC does not recommend a specific timeframe to enter into Phase 2, but assumes that after some suitable period with the basic signature mechanics in place, Phase 2 can be addressed and TUF implementation can be designed and added.
+
+## Cargo publish flow
+
+![publish flow](https://raw.githubusercontent.com/lulf/rfc-resources/main/sigstore/cargo_sigstore-publish.drawio.png)
+
+The following changes must be made to cargo when publishing a crate:
+
+1. Authenticate the publisher using OIDC to retrieve a token
+1. Generate the ephemeral signing key and certificate request
+1. Generate a certificate signing request and pass with a token to Sigstore Fulcio.
+1. Sign the generated .crate file using the private key
+1. Attach signature, identity and certificate to crates.io publish request
+1. Publish an entry to the Rekor log with the .crate digest and certificate public key
+
+Most of the above is already implemented in https://github.com/sigstore/sigstore-rs.
+
+### crates.io flow
+
+The crates.io sub-flow on publish is described below:
+
+![crates.io flow](https://raw.githubusercontent.com/lulf/rfc-resources/main/sigstore/cargo_sigstore-crates.io.drawio.png)
+
+The following components need changing for crates.io:
+
+1. Additional columns must be added to the PostgreSQL database
+1. HTTP API must handle a new version of NewCratePublish request with the signature data
+1. The HTTP handler talks to Rekor and verifies that the signature is valid and that owner matches certificate
```

Comment:
The RFC really doesn't do a great job of explaining the purpose of Rekor and its relationship to the protocol. After reading, I don't see what it does or why it's needed.

Reply:
Yes, I think this is leaking an implementation detail of how Sigstore's verifications work: the salient fact here isn't Rekor (which is an artifact transparency service), but that Crates is expected to verify the signature it's just received.